lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -88,43 +119,69 @@ class ReportUIFeatures {
   initFeatures(report) {
     this.json = report;
 
-    this._setupMediaQueryListeners();
-    this._dropDown.setup(this.onDropDownMenuClick);
-    this._setupThirdPartyFilter();
-    this._setupElementScreenshotOverlay(this._dom.find('.lh-container', this._document));
-    this._setUpCollapseDetailsAfterPrinting();
-    this._resetUIState();
-    this._document.addEventListener('keyup', this.onKeyUp);
-    this._document.addEventListener('copy', this.onCopy);
+    if (this.featureSet.mediaQueryListeners) {
+      this._setupMediaQueryListeners();
+    }
+
+    if (this.featureSet.dropDownMenu) {
+      this._dropDown.setup(this.onDropDownMenuClick);
+      this._document.addEventListener('copy', this.onCopy);
+    }
+
+    if (this.featureSet.thirdPartyFilter) {
+      this._setupThirdPartyFilter();
+    }
+
+    if (this.featureSet.elementScreenshotOverlay) {
+      let overlayContainerEl;
+      if (this.featureSet.elementScreenshotOverlay === true) {
+        overlayContainerEl = this._dom.find('.lh-container', this._document);
+      } else {
+        overlayContainerEl = this.featureSet.elementScreenshotOverlay.containerEl;
+      }
+      this._setupElementScreenshotOverlay(overlayContainerEl);
+    }
 
-    const topbarLogo = this._dom.find('.lh-topbar__logo', this._document);
-    topbarLogo.addEventListener('click', () => this._toggleDarkTheme());
+    if (this.featureSet.printing) {
+      this._setUpCollapseDetailsAfterPrinting();
+      this._document.addEventListener('keyup', this.onKeyUp);
+    }
+
+    this._resetUIState();
 
     let turnOffTheLights = false;
-    // Do not query the system preferences for DevTools - DevTools should only apply dark theme
-    // if dark is selected in the settings panel.
-    if (!this._dom.isDevTools() && window.matchMedia('(prefers-color-scheme: dark)').matches) {
-      turnOffTheLights = true;
+    if (this.featureSet.toggleDarkMode) {
+      const topbarLogo = this._dom.find('.lh-topbar__logo', this._document);
+      topbarLogo.addEventListener('click', () => this._toggleDarkTheme());
+
+      // Do not query the system preferences for DevTools - DevTools should only apply dark theme
+      // if dark is selected in the settings panel.
+      if (!this._dom.isDevTools() && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

Comment:
is the idea that a devtools rendering config would have `toggleDarkMode: false` (instead of this isDevTools()) ?

Reply:
didnt consider that but ya, sure